src/ossl/rsa.rs

```diff
@@ -431,6 +431,11 @@ impl RsaPKCSOperation {
                 return Err(e);
             }
         };
+        let needed_len = op.encryption_len(keydata.len(), true)?;
+        if output.len() < needed_len {
+            zeromem(keydata.as_mut_slice());
+            return Ok(needed_len);
+        }
```

Comment:
This should return Error::buf_too_small(needed_len);

Reply:
My bad. Fixed now!